src/components/inputs/Slider/Slider.tsx

```diff
@@ -35,50 +54,101 @@ export function Slider({
   range,
   ...props
 }: Props) {
-  const ref = useRef(null);
+  const trackRef = useRef(null);
+  const startHandleRef = useRef(null);
+  const endHandleRef = useRef(null);
+
+  useOutsideClick([trackRef], () => {
+    setFocus(null);
+  });
+
   const [state, dispatch] = useReducer(
     reducer,
     initialState(initialValues)
   );
+
   const { values, trackRect, focused, dragging }: State = state;
 
+  function dispatchChangeEvent(callback) {
+    const currentInput =
+      focused === 'startHandle' ? startHandleRef : endHandleRef;
+
+    const event = new Event('change', { bubbles: true });
+    currentInput?.current?.dispatchEvent(event);
+    callback && callback(event);
+  }
+
   function setDragging(payload) {
-    setFocus(payload);
+    if (payload) setFocus(payload);
     return dispatch({ type: 'SET_DRAGGING', payload });
   }
 
   function setValue(payload) {
+    dispatchChangeEvent(onChange);
     return dispatch({ type: 'SET_VALUES', payload });
   }
 
   function setFocus(payload) {
-    return () => dispatch({ type: 'SET_FOCUS', payload });
+    return dispatch({ type: 'SET_FOCUS', payload });
   }
 
   function setStartValue(value) {
-    const newValue = constrain(min, max)(value);
+    const newValue = constrain(
+      min,
+      range ? values[1] - 1 : max
+    )(value);
     setValue([newValue, values[1]]);
   }
 
   function setEndValue(value) {
-    const newValue = constrain(min, max)(value);
+    const newValue = constrain(values[0] + 1, max)(value);
     setValue([values[0], newValue]);
   }
 
   useLayoutEffect(() => {
-    const { left, width } = geometry(ref.current);
+    const { left, width } = geometry(trackRef.current);
     dispatch({ type: 'SET_TRACK_RECT', payload: { left, width } });
   }, [dragging]);
 
   useEffect(() => {
-    const mouseup = (event) => {
+    const mouseup = () => {
+      dispatchChangeEvent(onDragEnd);
       dragging && setDragging(false);
-      onDragEnd && onDragEnd(event);
+    };
+
+    const keydown = (event) => {
+      if (!focused || !isHandleFocused(focused)) return;
+
+      const action =
+        focused === 'startHandle' ? setStartValue : setEndValue;
+      const currentValue =
+        focused === 'startHandle' ? values[0] : values[1];
+
+      switch (event.keyCode) {
+        case arrowDown:
+        case arrowLeft:
+          action(currentValue - 1);
```

Comment:
There isn't a case in which we'd need to increment these by a different value aside from `1`, is there?

Reply:
Based on the specs [here](https://www.w3.org/WAI/ARIA/apg/patterns/slider/), it's either 1 or the max / min value.
Both flows are supported in this PR :)